api/types/kustomization.go

```diff
@@ -222,6 +223,21 @@ func (k *Kustomization) FixKustomizationPreMarshalling() error {
 	k.Patches = append(k.Patches, k.PatchesJson6902...)
 	k.PatchesJson6902 = nil
 
+	if k.PatchesStrategicMerge != nil {
+		for _, patchStrategicMerge := range k.PatchesStrategicMerge {
+			// check this patch is file path select.
+			if strings.Count(string(patchStrategicMerge), "\n") < 1 &&
+				(patchStrategicMerge[len(patchStrategicMerge)-5:] == ".yaml" || patchStrategicMerge[len(patchStrategicMerge)-4:] == ".yml") {
```

Comment:
An easier and more reliable way to check if this is a filepath or not is to try to open it (i.e. os.ReadFile), and check for error

Reply:
rewrite it.